src/catalogue.js

```diff
@@ -0,0 +1,119 @@
+import { first, pick, omit, get } from 'lodash';
+import * as xhr from './xhr';
+import Promise from 'bluebird';
+
+const BASIC_PROPS = ['title', 'summary', 'identifier', 'uri'];
+const REQUEST_DEFAULTS = {
+    types: ['attribute', 'metric', 'fact'],
+    paging: {
+        offset: 0,
+        limit: 'ALL'
+    }
+};
+
+function pickBasicProps(item) {
+    return pick(item.meta, BASIC_PROPS);
+}
+
+function deserializeFact(fact) {
+    return pickBasicProps(fact);
+}
+
+function deserializeMetric(metric) {
+    const basicProps = pickBasicProps(metric);
+    const metricProps = pick(metric.content, ['expression', 'format']);
+
+    return { ...basicProps, ...metricProps };
+}
+
+function deserializeAttribute(attribute) {
+    const basicProps = pickBasicProps(attribute);
+    const displayForm = first(attribute.content.displayForms);
+    const attributeProps = {
+        dimensionUri: get(attribute, ['content', 'dimension']),
+        granularity: get(attribute, ['meta', 'identifier']),
+        elementsUri: get(displayForm, ['links', 'elements']),
+        dfIdentifier: get(displayForm, ['meta', 'identifier']),
+        dfUri: get(displayForm, ['meta', 'uri']),
+        dateType: get(attribute, ['content', 'type'])
+    };
+
+    return { ...basicProps, ...attributeProps };
+}
+
+function deserializeDimension(dimension) {
+    const basicProps = pickBasicProps(dimension);
+    const attributes = get(dimension, ['content', 'attributes']);
+    const dimensionProps = {
+        attributes: attributes.map(deserializeAttribute)
+    };
+
+    return { ...basicProps, ...dimensionProps };
+}
+
+const DESERIALIZERS = {
+    fact: deserializeFact,
+    metric: deserializeMetric,
+    attribute: deserializeAttribute,
+    dimension: deserializeDimension
+};
+
+function deserializeItem(item) {
+    const type = first(Object.keys(item));
+    const props = DESERIALIZERS[type](item[type]);
```

Comment:
when we add more types some place else, this will blow up. check for existence of the deserializer so we at least have a decent error message.